clang/test/Driver/sycl-mlir.cpp

```diff
@@ -11,6 +11,27 @@
 // CHK-PHASES-LLVM: 2: compiler, {1}, ir, (device-sycl)
 // CHK-PHASES-LLVM: 3: offload, "device-sycl (spir64-unknown-unknown-syclmlir)" {2}, ir
 
+// RUN: %clangxx -ccc-print-phases --sysroot=%S/Inputs/SYCL             \
+// RUN: -target x86_64-unknown-linux-gnu -fsycl -c -fsycl-device-only   \
+// RUN: -fsycl-raise-host                                               \
+// RUN: -fsycl-targets=spir64-unknown-unknown-syclmlir %s 2>&1          \
+// RUN: | FileCheck -check-prefix=CHK-PHASES-LLVM-RAISE %s
+
+// CHK-PHASES-LLVM-RAISE: 0: input, "{{.*}}.cpp", c++, (host-sycl)
+// CHK-PHASES-LLVM-RAISE: 1: append-footer, {0}, c++, (host-sycl)
+// CHK-PHASES-LLVM-RAISE: 2: preprocessor, {1}, c++-cpp-output, (host-sycl)
```

Comment:
I was expecting to see that we start a device compilation (to produce the stub). Why do we have these 3 lines ?

Reply:
That's to produce the preprocessed file that will serve as an input to the compiler together with the stub file (built in phase 5 in this example)